### DISCUSSION TITLE
New Feature: Now Playing

### DISCUSSION THREAD

**Discussion by daniel-naegele** [oguzhnatly/flutter_carplay#15] [Q&A] [answered]
Hi,

is there an ETA for the "Now Playing" feature?

**Comment by oguzhnatly** [comment]
Hi @Butzlabben,

Unfortunately, there is no ETA for Now Playing feature. If you would like to contribute, you're always welcome to [Discord Server](https://discord.gg/Xz6WVezFfh).

Thank you,
Oguzhan

**Comment by snipd-mikel** [comment]
Hi @oguzhnatly. We are currently working on an integration with Carplay for our podcast app, so the Now playing feature is something we will definitely need. Because of that, I would like to contribute to this project with that feature. What are the steps to follow to contribute?

**Reply by neownd** [reply]
@oguzhnatly @snipd-mikel  Any progress on that? ;) Looking forward to it!

**Reply by subhangiP** [reply]
@oguzhnatly @snipd-mikel  I would like to know what is progress on the **Now Playing feature**. In our music app carplay feature is required. what is the ETA for the same? Looking forward to it!!!

**Comment by vanlooverenkoen** [comment]
We are using the `audio_service` package already broadcasts the now playing state. So if you use that package you have this already

**Reply by ghost** [reply]
@vanlooverenkoen when I run the example folder of the audio_service package in iOS it does not display the app in carplay. I have manually added the entitlements file and it gets crashed on launching the app from carplay. In Android, it is working fine but in iOS not.
Have u achieved the music template feature using the audio_service package in iOS?  please let me know the steps. Thanks!!!

**Reply by vanlooverenkoen** [reply]
Make sure you request the apple car play permission in the apple developer portal. For android there is no need for another permission.

After that you can create a orovisioning profile that contains the correct apple car play services

**Reply by ghost** [reply]
@vanlooverenkoen have u tested the audio_service package in iOS? In that pkg I didn’t see any code related to iOS carplay like flutter_Carplay has.

**Reply by vanlooverenkoen** [reply]
There is no iOS specific code for that. the ios audio player handles that internally for you

**Comment by vanlooverenkoen** [comment]
I created a pull request where it is easy to say. Got to the shared now playing template. 

(https://github.com/oguzhnatly/flutter_carplay/pull/22)

**Comment by oguzhnatly** [comment, marked as answer]
Now Playing is actively in development! Expect a PR very soon. Thanks for the patience everyone 🙏
